src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1262,6 +1272,31 @@ func (k *kataAgent) createContainer(ctx context.Context, sandbox *Sandbox, c *Co
 	// passing irrelevant information to the agent.
 	k.constrainGRPCSpec(grpcSpec, passSeccomp, sandbox.config.VfioMode == config.VFIOModeGuestKernel)
 
+	if sandbox.config.HypervisorType == RemoteHypervisor {
+		// Need to send a PullImage request for "pause" image before CreateContainer request to ensure the image
+		// is downloaded inside the VM. Ideally this should be the case for all hypervisors when using CC
+		// TODO: remove this code when containerd supports the new Sandbox API
+
+		image, err := getImageName(ociSpec.Annotations)
+		if err != nil {
+			k.Logger().Infof("no image spec available: %v", err)
+			return nil, err
+		}
+
+		//Send PullImageRequest only for "pause" image
+		if image == "k8s.gcr.io/pause:3.2" {
```

Comment:
@yoheiueda for crio we'll need to pull all the images including pause. This ensures no changes to the crio side. Can we enable pullImageRequest for all the images in case of remote hypervisor ?

Reply:
I am going to implement image pulling in the cloud-api-adaptor side. https://github.com/confidential-containers/cloud-api-adaptor/issues/88

If cri-o provides image name in annotations, cloud-api-adaptor will be able to send pull image requests for all container images including pause.

The current code to get container image name in kata shim is this one.
https://github.com/yoheiueda/kata-containers/blob/8f176024bcd7d78b81439df75824e068b31eaddd/src/runtime/virtcontainers/kata_agent.go#L1316-L1338

For some reason, containerd v1.6.6-15 does not provide annotation for image name, so we still need the custom conatinerd.

@bpradipt do you know whether cri-o provide image name as annotation `io.kubernetes.cri-o.ImageName`?

Reply:
@yoheiueda yes, the annotation is provided. 
This is from a running system
```
        "io.kubernetes.cri-o.ImageName": "quay.io/karmab/origin-coredns:latest",
        "io.kubernetes.cri-o.Name": "k8s_coredns_coredns-test-worker-0.ocp411.com_openshift-infra_dbbb729d580e7461e98ebf6dba93ab7c_1",
        "io.kubernetes.cri-o.ContainerType": "container",
        "io.kubernetes.cri-o.Stdin": "false",
```

Reply:
I implemented https://github.com/confidential-containers/cloud-api-adaptor/issues/88, and remove the work around from kata shim.

I also added the cri-o support.